superflore/generators/bitbake/run.py

```diff
@@ -76,6 +77,24 @@ def main():
         warn('"{0}" distro detected...'.format(args.ros_distro))
         selected_targets = [args.ros_distro]
         preserve_existing = False
+    # open cached tar file if it exists
+    md5_cache = None
+    sha256_cache = None
+    if args.tar_archive_dir:
+        try:
+            md5_file = open('%s/md5_cache.pickle' % args.tar_archive_dir, 'rb')
```

Comment:
I recommend using a context manager here to open and close the file automatically.

https://jeffknupp.com/blog/2016/03/07/python-with-context-managers/